src/main/java/com/houarizegai/calculator/ui/CalculatorUI.java

```diff
@@ -505,6 +508,19 @@ private void initButtons(int[] columns, int[] rows) {
             }
         });
         btnLog.setVisible(false);
+
+        btnPi = createButton("π", columns[4], rows[4]);
+        btnPi.addActionListener(event -> {
+            if (addToDisplay) {
+                inputScreen.setText(String.valueOf(Math.PI));
+                addToDisplay = false;
+            } else {
+                inputScreen.setText(String.valueOf(Math.PI));
+                addToDisplay = true;
+            }
+            go = true;
```

Comment:
**Issue**: The action listener for the btnPi button toggles the addToDisplay boolean unnecessarily and sets the inputScreen text to Math.PI regardless of the addToDisplay state. <br> **Fix**: Simplify the action listener logic by directly setting the inputScreen text to Math.PI without toggling the addToDisplay boolean. <br> **Code Suggestion**: 
 ```
 -            if (addToDisplay) {
 -                inputScreen.setText(String.valueOf(Math.PI));
 -                addToDisplay = false;
 -            } else {
 -                inputScreen.setText(String.valueOf(Math.PI));
 -                addToDisplay = true;
 -            }
 +            inputScreen.setText(String.valueOf(Math.PI));
 ```